### ISSUE TITLE
Error during installation completion (Route [voyager.voyager_extension_assets] not defined.)

### ISSUE THREAD

**Issue by locopine** [MonstreX/voyager-extension#42]
Very interesting proposal! My congratulations to the developer.

Too bad that due to the error, as shown below, where it is presented right after installing the package and nothing else in "php artisan" works because of the failure of the route, I could not test it.

`18  artisan:37 `
`      Illuminate\Foundation\Console\Kernel::handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) `
` `
`   Symfony\Component\Routing\Exception\RouteNotFoundException `
` `
`  Route [voyager.voyager_extension_assets] not defined. `
` `
`  at vendor/laravel/framework/src/Illuminate/Routing/UrlGenerator.php:466 `
`    462▕         if (! is_null($route = $this->routes->getByName($name))) { `
`    463▕             return $this->toRoute($route, $parameters, $absolute); `
`    464▕         } `
`    465▕  `
`  ➜ 466▕         throw new RouteNotFoundException("Route [{$name}] not defined."); `
`    467▕     } `
`    468▕  `
`    469▕     /** `
`    470▕      * Get the URL for a given route instance. `
` `
`      +11 vendor frames `
`  12  [internal]:0 `
`      Illuminate\Foundation\Application::Illuminate\Foundation\{closure} `
` (Object(MonstreX\VoyagerExtension\VoyagerExtensionServiceProvider)) `

This is due to the function call/check in the file "vendor/monstrex/voyager-extension/src/Helpers/helpers.php" on line '90'.

Any suggestions on how to solve the problem? I have tried several alternatives, but without success so far.

Much obliged.

**Comment by MonstreX** [comment]
Thank you!
I've just checked it out (on the fresh installed laravel 9+ and voyager 1.5). I have not seen any mentioned errors. How can I reproduce this behavior? Have you done all the installation steps?

**Comment by Metilar** [comment]
More closely examined, the problem appears after the command `artisan route:cache`

After that, `VoyagerExtensionServiceProvider` in the line
` $this->loadRoutesFrom(__DIR__ . '/../routes/routes.php');`
Routes are not registered, they are in the cache allegedly.

**Solution 1**: comment out these lines, run `artisan route:clear` and uncomment the lines.
But the result is that you can't cache routes =(

```
Config::set(
	'voyager.additional_css',
	array_merge(config('voyager.additional_css'), [voyager_extension_asset('css/app.css')])
);

Config::set(
	'voyager.additional_js',
	array_merge(config('voyager.additional_js'), [voyager_extension_asset('js/app.js')])
);
```

**Update - Solution 2**: override the function, as a result, it is possible to cache routes.
```
protected function loadRoutesFrom($path)
{
    require $path;
}
```

**Comment by MonstreX** [comment]
> **Update - Solution 2**: override the function, as a result, it is possible to cache routes.  

Thank you a lot! Seems it really works. Will be fixed on the next release.

**Comment by Metilar** [comment]
> Thank you a lot! Seems it really works. Will be fixed on the next release.

This certainly works, but I'm not sure if this is the right solution, if I understand correctly then it turns out that these particular routes will not be cached

**Comment by locopine** [comment]
> > **Update - Solution 2**: override the function, as a result, it is possible to cache routes.
> 
> Thank you a lot! Seems it really works. Will be fixed on the next release.

The following error occurs when trying to remove an image:

```
voyager-extension-assets?path=js%2Fapp.js:2 Uncaught ReferenceError: vext_page_slug is not defined
    at HTMLAnchorElement.<anonymous> (voyager-extension-assets?path=js%2Fapp.js:2:54276)
    at HTMLDivElement.dispatch (voyager-assets?path=js%2Fapp.js:2:1190520)
    at g.handle (voyager-assets?path=js%2Fapp.js:2:1188515)
```
Removing the image removal extension returns to normal, although the response of the new "src's" addresses of the images no longer work!

The proposal is excellent, but unfortunately I will not be able to use it in production!

Much obliged.

**Comment by MonstreX** [comment]
> Removing the image removal extension returns to normal, although the response of the new "src's" addresses of the images no longer work!

How can I reproduce this behavior?

**Comment by locopine** [comment]
> > Removing the image removal extension returns to normal, although the response of the new "src's" addresses of the images no longer work!
> 
> How can I reproduce this behavior?

I arrived at the real cause for what happened and, being able to elude myself how I can proceed to avoid the error in my use case, I thank you.

What happens is the following:
- I created a new table and from that table (products) a BREAD (basic) from the created table;
- the BREAD created, only by Voyager (default) works perfectly fine; already with the extension only this new BREAD is not recognized.

In short, on Post, and Page the extension works perfectly fine; but why doesn't the extension work on all Voyager?

Much obliged.

PS.: Additional information:
- the behavior described above occurred because I was overriding BREAD views! Curious... I believe I shouldn't have this behavior.
I adapted it briefly for use with BREAD without replacement and the extension worked in its entirety.

**Comment by MonstreX** [comment]
Yes, you have to be very careful when overriding objects like View. The extension inside overrides some original Voyager views so some other your own overrides may not work correctly or as you assumed.
